whelktool/scripts/syslib/ISNI.groovy

```diff
@@ -0,0 +1,43 @@
+PrintWriter failedAuthIDs = getReportWriter("failed-authIDs")
+PrintWriter scheduledForUpdating = getReportWriter("scheduled-updates")
+
+def listofIds = []
+def isnimap = [:]
+
+new File(scriptDir, "isni_numbers.txt").each{ row -> 
+
+def (ID, ISNI) = row.split(/\t/).collect { it.trim() } // hämta id och nummer, separera på tab
+
+listofIds.add(ID)
+isnimap[ID] = ISNI 
+}
+
+def ids = selectByIds(listofIds) { auth ->
+
+def aukt = auth.graph[1]
+def uri = auth.graph[0]["@id"]
+
+id = uri.split('/')[3] // ta ut id från uri
+
+def ISNI = isnimap[id] // matcha ISNI-nummer mot id
+
+aukt["identifiedBy"]?.removeIf { it["typeNote"]?.contains("isni") || false } // radera element i identifiedBy som har ISNI-nummer (ofta felaktiga - dessa kan iofs vara korrekta, men läses in på nytt i senare steg))
```

Comment:
This looks wrong to me!

it["typeNote"]?.contains("isni") reasonably returns true or false (or null which groovy considers false)

{ true or false } || false is, redundant at best. "OR false" is by definition a No-op.

You should probably remove the "|| false" part!

Reply:
Thanks!

Reply:
Actually, without || false, it will throw a NullPointer exception if it returns null. So added back again.